tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -1035,3 +1037,68 @@ def process(self, resources, event=None):
         parent_resources_ids = [p['id'] for p in parent_resources]
         parent_key = self.manager.resource_type.parent_key
         return [r for r in resources if r[parent_key] in parent_resources_ids]
+
+
+class AzureAdvisorFilter(RelatedResourceFilter):
+    """
+    Filter resources by Azure Advisor Recommendations
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: disks-with-cost-recommendations
+            resource: azure.disk
+            filters:
+              - type: advisor-recommendation
+                key: '[].properties.category'
+                value: Cost
+                value_type: swap
+                op: in
+
+    """
+
+    RelatedResource = "c7n_azure.resources.advisor.AdvisorRecommendation"
+    RelatedIdsExpression = "id"
+    AnnotationKey = "AdvisorRecommendation"
+
+    _recommendation_map = {}
```

Comment:
switch class variable to non mutable default value Ala None, (), usage is an instance variable now.